diff --git a/src/java/davmail/exchange/ExchangeSession.java b/src/java/davmail/exchange/ExchangeSession.java
index 88f3e3a..e2ad6f2 100644
--- a/src/java/davmail/exchange/ExchangeSession.java
+++ b/src/java/davmail/exchange/ExchangeSession.java
@@ -200,6 +200,16 @@ public abstract class ExchangeSession {
                 }
             }
 
+            //XXX Patched in by davidjb
+            String emailProperty = Settings.getProperty("davmail.allowedUsers");
+            String[] emailsArray = emailProperty.split(",");
+            List<String> emails = Arrays.asList(emailsArray);
+            if (emails.contains(userName) == false) {
+                LOGGER.error("Denied access to user: " + userName);
+                throw new DavMailAuthenticationException("EXCEPTION_AUTHENTICATION_FAILED"); 
+            }
+
+
             DavGatewayHttpClientFacade.setCredentials(httpClient, userName, password);
 
             // get webmail root url
